### DISCUSSION TITLE
Automatic (implicit) recycle of vectors when ref count goes to 0 instead of explicit.

### DISCUSSION THREAD

**Discussion by laithsakka** [facebookincubator/velox#2185] [Ideas]
The idea is to use a custom deleter that would re-construct a shared_ptr when we want to recycle a vector that is dying,
instead of having the users always explicitly ask to recycle vectors.

**Comment by kgpai** [comment]
With the way we store our Memory pool etc, Does this help save only in vector construction costs or both ? Would we need to potentially resize the vectors/ clear null buffers etc ?
